bundles/framework/maplegend/MapLegendList.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Collapse } from 'antd';
+import { InfoCircleOutlined } from '@ant-design/icons';
+import styled from 'styled-components';
+
+import 'antd/es/collapse/style/index.js';
+
+const { Panel } = Collapse;
+
+const MetadataIcon = styled(InfoCircleOutlined)`
+    & {
+        margin: 0 0 0 10px;
+    }
+`;
+
+export const MapLegendList = ({ legendList, noImageText }) => {
+    const [listState, setListState] = useState({
+        list: legendList
+    });
+
+    const composeHeader = (title, uuid, callback) => {
+        return (
+            <Fragment>
+                { title }
+                { uuid && <MetadataIcon onClick={ (event) => callback(event, uuid) } /> }
+            </Fragment>
+        );
+    };
+
+    const composeLegendImage = (item, index) => {
+        return (
+            <img
+                onError={ () => setListState(listState => {
```

Comment:
Using state feels a bit of an overkill for this. Could we just add a class for the img on error that hides it and shows the message instead? Maybe it could be it's own component even under oskariui? Or atleast have the state for the img instead of the whole list?

Reply:
Could try something like this:
```
css:
.maplegend img.errored {
    display:none;
}
.maplegend img + span {
    display: none;
}
.maplegend img.errored + span {
    display: block;
}
jsx: 
<img src={url} onError={() => this.classList.add("errored")} />
<Message messageKey='invalidLegendUrl' />
```
1) Message is wrapped as span by default
2) By default a span following the img is hidden
3) If the img triggers onError it gets the class "errored"
4) A span followed by img.errored will be shown

It's not ideal as the message will be on the DOM either way but using this just feels a bit iffy here. Maybe it would be a better way to create the common component for images that use state internally to show another component (Message here) instead on error?

Reply:
Made a component for image and added own state for it. Now we don't add the whole list into state anymore.